java/code/src/org/keyczar/KeyczarKey.java

```diff
@@ -57,6 +58,8 @@
   private static final int PBE_SALT_SIZE = 8;
   private static final int IV_SIZE = 16;
   private static final int PBE_ITERATION_COUNT = 1000;
+  
+  protected ArrayList<byte[]> fallbackHash = new ArrayList<byte[]>();
```

Comment:
It looks like hashes are defined per Key type. What is the reason for that and why doesn't fallbackHash belong in the same place?